src/eval_framework/file_utils/file_ops.py

```diff
@@ -0,0 +1,173 @@
+
+        try:
+            artifact_path = artifact.download(root=str(self.download_path))
+            wandb.use_artifact(artifact)
+        except Exception as e:
+            # patch the wandb boto3 call to disable ssl verification
+            try:
+                print(f"failed to download artifact {e}, patching boto3")
+                with patch(
+                    "boto3.session.Session.resource",
+                    lambda _, *args, **kwargs: boto3.session.Session().resource(*args, **kwargs, verify=False),
+                ):
+                    with warnings.catch_warnings():
+                        # this is to suppress the insecure request warning from urllib3
+                        # the attribute exists, but mypy cannot resolve it
+                        warnings.simplefilter(
+                            "ignore",
+                            category=requests.packages.urllib3.exceptions.InsecureRequestWarning,  # type: ignore
+                        )
+                        artifact_path = artifact.download(root=str(self.download_path))
+                wandb.use_artifact(artifact)
+            except Exception as e:
+                print(f"failed to download artifact {e}, please check your AWS credentials and endpoint")
+                raise e
+
+        return Path(artifact_path)
+
+    def find_hf_checkpoint_root_from_path_list(self) -> str | None:
+        """Find HuggingFace checkpoint root from a list of file paths.
+
+        Args:
+            file_paths: List of file paths (can be S3 URIs or local paths)
+
+        Returns:
+            str | None: Path to the HuggingFace checkpoint root folder, or None if not found
+        """
+        download_path = (
+            self.download_path.name
+            if isinstance(self.download_path, tempfile.TemporaryDirectory)
+            else self.download_path
+        )
+
+        if not download_path:
+            return None
+
+        checkpoint_roots = [x for x in Path(download_path).glob("**/config.json")]
+        if checkpoint_roots:
+            assert len(checkpoint_roots) == 1, (
+                "Multiple checkpoints found"
+            )  # if there are more than one, we have a problem
+            return str(checkpoint_roots[0].parent)
+
+        return None
+
+    def cleanup(self) -> None:
+        self._cleanup_temp_dir()
+
+    def __enter__(self) -> "WandbFs":
+        return self
+
+    def __exit__(self, exc_type: Any, exc_val: Any, exc_tb: Any) -> None:
+        self._cleanup_temp_dir()
+
+    def _cleanup_temp_dir(self) -> None:
+        if hasattr(self, "_temp_dir") and self._temp_dir:
+            try:
+                self._temp_dir.cleanup()
+            except (OSError, FileNotFoundError):
+                # Directory might already be cleaned up or removed
+                pass
+            finally:
+                self._temp_dir = None
+                self.download_path = None
+
+    def __del__(self) -> None:
```

Comment:
I'd suggest to remove this, __del__ is nearly never useful in practice. If the class is used as context manager, it's handled. If not, _setup_cleanup_handlers() deals it with this.